vassal-app/src/main/java/VASSAL/tools/image/LabelUtils.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2000-2020 by Rodney Kinney, Joel Uckelman, Brent Easton
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License (LGPL) as published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, copies are available
+ * at http://www.opensource.org.
+ */
+
+package VASSAL.tools.image;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+
+public class LabelUtils {
+  private LabelUtils() {}
+
+  public static final int CENTER = 0;
+  public static final int RIGHT = 1;
+  public static final int LEFT = 2;
+  public static final int TOP = 3;
+  public static final int BOTTOM = 4;
+
+  public static void drawLabel(Graphics g, String text, int x, int y, int hAlign, int vAlign, Color fgColor, Color bgColor) {
+    drawLabel(g, text, x, y, new Font("Dialog", Font.PLAIN, 10), hAlign, vAlign, fgColor, bgColor, null);
+  }
+
+  public static void drawLabel(Graphics g, String text, int x, int y, Font f, int hAlign, int vAlign, Color fgColor, Color bgColor, Color borderColor) {
+    g.setFont(f);
+    final int width = g.getFontMetrics().stringWidth(text + "  ");
+    final int height = g.getFontMetrics().getHeight();
+    int x0 = x;
+    int y0 = y;
+
+    switch (hAlign) {
+    case CENTER:
+      x0 = x - width / 2;
+      break;
+    case LEFT:
+      x0 = x - width;
+      break;
+    }
+
+    switch (vAlign) {
+    case CENTER:
+      y0 = y - height / 2;
+      break;
+    case BOTTOM:
+      y0 = y - height;
+      break;
+    }
+
+    if (bgColor != null) {
+      g.setColor(bgColor);
+      g.fillRect(x0, y0, width, height);
+    }
+
+    if (borderColor != null) {
+      g.setColor(borderColor);
+      g.drawRect(x0, y0, width, height);
+    }
+
+    g.setColor(fgColor);
+    ((Graphics2D) g).setRenderingHint(RenderingHints.KEY_ANTIALIASING,
+                                      RenderingHints.VALUE_ANTIALIAS_ON);
+    g.drawString(" " + text + " ", x0,
+      y0 + g.getFontMetrics().getHeight() - g.getFontMetrics().getDescent());
```

Comment:
Should this have g.addRenderingHints(SwingUtils.FONT_HINTS); to ensure the Rendering hints are in a known state?

Reply:
Done.